File: silver/level_3/1449.py
# ver.1  84ms
import sys
input = sys.stdin.readline

N,L = map(int, input().split())
loc = list(map(int, input().split()))
loc.sort()

checked = [False]*N; cnt = 0; 
for i in range(N):
  if not checked[i]:
    checked[i] = True
    cnt += 1
    temp = loc[i] + L - 1
    for j in range(i+1,N):
      if loc[j] <= temp:
        checked[j] = True
print(cnt)


# ver.2  88ms
import sys
input = sys.stdin.readline

N,L = map(int, input().split())
loc = list(map(int, input().split()))
loc.sort()

cnt = 1; end = loc[0] + L - 1; 
for element in loc:
  if element <= end:
    continue
  cnt += 1
  end = element + L - 1
print(cnt)

File: baekjoon/2_silver/level_3/cpp/1449.cpp
// ver.1  0ms
#include <iostream>
#include <algorithm>
using namespace std;
#define MAX_NUM 1001

int N, L, cnt, temp;
int loc[MAX_NUM];
bool checked[MAX_NUM];

int main() {
  ios::sync_with_stdio(false);
  cin.tie(NULL); cout.tie(NULL);

  cin >> N >> L;
  for (int i = 0; i < N; i++) {
    cin >> loc[i];
  }
  sort(loc, loc+N);

  for (int i = 0; i < N; i++) {
    if (!checked[i]) {
      cnt++;
      temp = loc[i] + L - 1;
      for (int j = i+1; loc[j] <= min(temp, MAX_NUM); j++) {
        checked[j] = true;
      }
    }
  }
  cout << cnt;

  return 0;
}
